pylint/checkers/classes/class_checker.py

```diff
@@ -1298,6 +1298,14 @@ def form_annotations(arguments):
                 if called_annotations != overridden_annotations:
                     return
 
+            if (
+                function.returns is not None
+                and meth_node.returns is None
+                or meth_node.returns.as_string() != function.returns.as_string()
+            ):
```

Comment:
```suggestion
                if function.returns is not None and (
                    meth_node.returns is None
                    or meth_node.returns.as_string() != function.returns.as_string()
                ):
```
To fix the tests, you would need to use the old condition.

For my https://github.com/PyCQA/pylint/pull/6141#issuecomment-1099627809, I suggested something a little bit different.
```suggestion
            if (
                function.returns is not None
                and meth_node.returns is not None
                and meth_node.returns.as_string() != function.returns.as_string()
            ):
```